panic: runtime error: hash of unshashble type multierror.chain

Something about `.addExceptionsFromError` doesn't seem to play nice with error chain:

```
panic: runtime error: hash of unhashable type multierror.chain

goroutine 1 [running]:
github.com/TheZeroSlave/zapsentry.(*core).addExceptionsFromError(0xc000030be0, {0xc000030c30, 0x0, 0x1}, 0xc00016c870, {0xf25f20, 0xc00062c820})
        /home/runner/go/pkg/mod/github.com/!the!zero!slave/zapsentry@v1.9.0/core.go:152 +0xf1
github.com/TheZeroSlave/zapsentry.(*core).createExceptions(0xc000030be0)
        /home/runner/go/pkg/mod/github.com/!the!zero!slave/zapsentry@v1.9.0/core.go:126 +0x19c
github.com/TheZeroSlave/zapsentry.(*core).Write(0xc000031b30, {0x2, {0xc07c2f23013d5cb7, 0x38836a63, 0x1505080}, {0x0, 0x0}, {0xe18e3b, 0x11}, {0x1, ...}, ...}, ...)
        /home/runner/go/pkg/mod/github.com/!the!zero!slave/zapsentry@v1.9.0/core.go:94 +0x56e
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc00064fec0, {0xc00061be00, 0x1, 0x1})
        /home/runner/go/pkg/mod/go.uber.org/zap@v1.20.0/zapcore/entry.go:220 +0x26d
go.uber.org/zap.(*Logger).Error(0xc000238ae0, {0xe18e3b, 0x11}, {0xc00061be00, 0x1, 0x1})
        /home/runner/go/pkg/mod/go.uber.org/zap@v1.20.0/logger.go:208 +0x76
github.com/zllovesuki/t/server.(*Server).Gossip(0xc000122500)
        /home/runner/work/t/t/server/gossip.go:32 +0x252
main.main()
        /home/runner/work/t/t/cmd/server/main.go:294 +0x363b
```

The culprit seems to be this https://github.com/TheZeroSlave/zapsentry/blob/051a995f4e844a68c74f72a7c7e0782964260ab8/core.go#L152

--- a/core.go
+++ b/core.go
@@ -119,7 +119,7 @@ func (c *core) createExceptions() []sentry.Exception {
 		return nil
 	}
 
-	processedErrors := make(map[error]struct{}, errorsCount)
+	processedErrors := make(map[string]struct{}, errorsCount)
 	exceptions := make([]sentry.Exception, 0, errorsCount)
 
 	for i := errorsCount - 1; i >= 0; i-- {
@@ -143,17 +143,21 @@ func (c *core) createExceptions() []sentry.Exception {
 	return exceptions
 }
 
+func getTypeOf(err error) string {
+	return err.Error() + reflect.TypeOf(err).String()
+}
+
 func (c *core) addExceptionsFromError(
 	exceptions []sentry.Exception,
-	processedErrors map[error]struct{},
+	processedErrors map[string]struct{},
 	err error,
 ) []sentry.Exception {
 	for i := 0; i < maxErrorDepth && err != nil; i++ {
-		if _, ok := processedErrors[err]; ok {
+		if _, ok := processedErrors[getTypeOf(err)]; ok {
 			return exceptions
 		}
 
-		processedErrors[err] = struct{}{}
+		processedErrors[getTypeOf(err)] = struct{}{}
 
 		exception := sentry.Exception{Value: err.Error(), Type: reflect.TypeOf(err).String()}
 

